eng/build-job.yml

```diff
@@ -50,6 +51,19 @@ jobs:
       - script: set __TestIntermediateDir=int&&build.cmd $(buildConfig) $(archType) -skiptests
         displayName: Build product
 
+    # Sign on Windows
+    - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), eq(parameters.osGroup, 'Windows_NT')) }}:
```

Comment:
Since we're only checking Build.Reason against one string, let's use `ne` instead of `notin`. I'm updating the other uses of `notin` in https://github.com/dotnet/coreclr/pull/21536 for consistency.